source/Planet.cpp

```diff
@@ -231,6 +233,28 @@ void Planet::Load(const DataNode &node)
 
 
 
+void Planet::FinishLoading(Set<Wormhole> &wormholes)
+{
+	// If this planet is in multiple systems, then it is a wormhole.
+	if(!wormhole && systems.size() > 1)
+	{
+		wormhole = wormholes.Get(name);
+		wormhole->LoadFromPlanet(this);
+		Files::LogError("Warning: auto generation of wormhole \"" + name
+				+ (wormhole->IsValid() ? "\"." : "\" failed."));
+	}
+
+	// If the wormhole was autogenerated we need to update it to
+	// match the planet's state.
+	if(wormhole && wormhole->IsAutogenerated())
```

Comment:
Why the `IsAutogenerated()` tracking and the separate if-statement? The only situation in which `IsAutoGenerated()` would be true is if LoadFromPlanet() was used to create the wormhole, but the only place where that happens is on line 242 just above here. So we can handle all the effects already above (we could even do the check if the description is empty within the LoadFromPlanet function and handle the GenerateLinks also within the LoadFromPlanet).

Reply:
`FinishedLoading` can be called multiple times. If a old-school wormhole (aka a planet) is updated, then we also need to update its wormhole object.